velox/core/QueryCtx.h

```diff
@@ -26,16 +26,20 @@
 
 namespace facebook::velox::core {
 
-class QueryCtx {
+class QueryCtx : public std::enable_shared_from_this<QueryCtx> {
```

Comment:
Can we separate the sp refactoring as a stand-alone one? It touches many files and would be cleaner if done by itself.

Reply:
This is not refactoring and is required for this PR.